app/webpacker/components/Results/ResultsFilter.jsx

```diff
@@ -0,0 +1,100 @@
+import React, { useMemo } from 'react';
+import {
+  Button, ButtonGroup, Form, Segment,
+} from 'semantic-ui-react';
+import _ from 'lodash';
+import { EventSelector } from '../wca/EventSelector';
+import { RegionSelector } from '../CompetitionsOverview/CompetitionsFilters';
+import { countries } from '../../lib/wca-data.js.erb';
+import I18n from '../../lib/i18n';
+
+export default function ResultsFilter({
+  filterState, filterActions, showCategories, isRecords,
+}) {
+  const {
+    event,
+    region,
+    rankingType,
+    gender,
+    show,
+  } = filterState;
+
+  const {
+    setEvent,
+    setRegion,
+    setRankingType,
+    setGender,
+    setShow,
+  } = filterActions;
+
+  const regionIso2 = useMemo(() => {
+    if (region === 'world') {
+      return 'all';
+    }
+    return countries.real.find((country) => country.id === region)?.iso2 || region;
+  }, [region]);
+  return (
+    <Segment raised>
+      <Form>
+        <Form.Field>
+          <EventSelector
+            title={I18n.t('results.selector_elements.events_selector.event')}
+            selectedEvents={[event]}
+            onEventSelection={({ eventId }) => setEvent(eventId)}
+            hideAllButton={!isRecords}
+            hideClearButton
+          />
+        </Form.Field>
+        <Form.Field>
+          <RegionSelector
+            region={regionIso2}
+            dispatchFilter={({ region: r }) => {
+              if (r === 'all') {
+                setRegion('world');
+              } else {
+                setRegion(countries.byIso2[r]?.id ?? r);
+              }
```

Comment:
Please try to avoid inlining more than one line in method body of JSX return components.

Reply:
For example, you could extract the `region` value somewhere (I have a feeling that it already ought to be extracted?) and just call `setRegion(someMemoedRegion)` as a one-liner

Reply:
well the correct solution would be to not use ids in the backend and iso2s in the frontend (except when we use Ids for continents..............), but that would require significant refactoring. I introduced a getRegionIdWithFallback method for this